scripts/gen_hashed_bin.py

```diff
@@ -117,32 +130,77 @@ def get_args():
 		help='The input tee_pageable.bin')
 
 	parser.add_argument('--out', \
-		required=True, type=argparse.FileType('wb'), \
+		required=False, type=argparse.FileType('wb'), \
 		help='The output tee.bin')
 
+	parser.add_argument('--out_header_v2', \
+		required=False, type=argparse.FileType('wb'), \
+		help='The output tee_header_v2.bin')
+
+	parser.add_argument('--out_pager_v2', \
+		required=False, type=argparse.FileType('wb'), \
+		help='The output tee_pager_v2.bin')
+
+	parser.add_argument('--out_pageable_v2', \
+		required=False, type=argparse.FileType('wb'), \
+		help='The output tee_pageable_v2.bin')
+
 	return parser.parse_args();
 
 def main():
 	args = get_args()
 	init_bin_size	   = args.init_size
 	tee_pager_fname	   = args.tee_pager_bin
 	tee_pageable_fname = args.tee_pageable_bin
-	outf		   = args.out
-
-	write_header(outf, 0, args, 0)
-	header_size = outf.tell();
-	append_to(outf, 0, tee_pager_fname)
-	append_to(outf, 0, tee_pageable_fname, init_bin_size)
-	append_hashes(outf, tee_pageable_fname)
-	init_size = outf.tell() - header_size;
-	append_to(outf, init_bin_size, tee_pageable_fname)
-	paged_size = outf.tell() - init_size - header_size;
-
-	outf.seek(0)
-	#print "init_size 0x%x" % init_size
-	write_header(outf, init_size, args, paged_size)
-
-	outf.close()
+	header_v1_size     = get_header_v1_size();
+	pager_input_size   = os.path.getsize(tee_pager_fname);
+	paged_input_size   = os.path.getsize(tee_pageable_fname);
+	hash_size          = paged_input_size / (4 * 1024) * \
+			     hashlib.sha256().digest_size
+
+	if paged_input_size % (4 * 1024) != 0:
+		print("Error: pageable size not a multiple of 4K:" + \
+			repr(paged_input_size))
+		sys.exit(1)
+
+	init_size	   = pager_input_size
```

Comment:
My fault: discard my comment....

Maybe it could be more clear to simply compute:
```python
init_size	  = pager_input_size + min(init_bin_size, paged_input_size) + hash_size
paged_size	  = paged_input_size + min(init_bin_size, paged_input_size)

```